frontend/src/components/UserProfile/UserProfile.tsx

```diff
@@ -143,7 +166,18 @@ export const _UserProfile = ({
           </div>
 
           <Divider />
-          <div className="my-4 text-center">Latest Activity</div>
+          {loadingUserData ? (
```

Comment:
Typically we use action verbs in loading states.
`isLoadingUserData` more readable than `loadingUserData`

Reply:
done